### PR TITLE
ConfigTree: Finish

### PR DESCRIPTION
**Pull request by chleh**
This PR cleans up the remains of the old ConfigTree and adds some tests for the new ConfigTree. 

The forward template declaration of boost ptree is still in ConfigTree.h/cpp.

This PR again affects many lines, however, most of the changes are mere refactorings. Only the 4th commit introduces a few lines of code. I think, it's best to read commit-wise.

### CODE REVIEW

**Comment by endJunction** [comment]
perfect. :fast_forward:

Because of non-mixed changes (variable renames, file name changes, separation of features) it was really easy to oversee the important changes and review. Thanks!

File: Tests/BaseLib/TestConfigTree.cpp [hunk 1]
```diff
@@ -220,12 +216,19 @@ TEST(BaseLibConfigTree, Get)
...
-            RUN_SAFE(z.getConfAttribute<double>("not_an_attr")); // nonexistent attribute
+            EXPECT_ANY_THROW(z.getConfAttribute<double>("not_an_attr")); // nonexistent attribute
             EXPECT_ERR_WARN(cbs, true, false);
             EXPECT_EQ(32.0, z.getValue<double>());
             EXPECT_ERR_WARN(cbs, false, false);
+            auto const opt = z.getConfAttributeOptional<bool>("optattr");
+            EXPECT_TRUE(!!opt); EXPECT_FALSE(*opt);
+            EXPECT_ERR_WARN(cbs, false, false);
+            //EXPECT_ANY_THROW(z.getConfAttributeOptional<bool>("optattr")); // getting attribute twice
+            //EXPECT_ERR_WARN(cbs, true, false);
```
Comment (TomFischer) [inline Tests/BaseLib/TestConfigTree.cpp]: Why are this lines commented?

**Comment by TomFischer** [comment]
:fast_forward:

**Comment by bilke** [comment]
:+1:

File: Tests/BaseLib/TestConfigTree.cpp (hunk 1 above)
Comment (chleh) [inline Tests/BaseLib/TestConfigTree.cpp]: Uncommented the lines.

**Pull request merged**